daemon/graphdriver/btrfs/btrfs_test.go

```diff
@@ -172,100 +133,60 @@ func TestBtrfsSubvolLongPath(t *testing.T) {
 
 	for i, l := 1, len(subvoldir); l <= syscall.PathMax*2; i++ {
 		dfile, err := os.OpenFile("dummyFile", os.O_RDONLY|os.O_CREATE, 0666)
-		if err != nil {
-			t.Fatalf("Failed to create file at %s: %v", subvoldir, err)
-		}
-		if err := dfile.Close(); err != nil {
-			t.Fatal(err)
-		}
+		assert.NilError(t, err, "Failed to create file at %s", subvoldir)
+		assert.NilError(t, dfile.Close())
 		name := getMaxFilenameFormPattern(fmt.Sprintf("LongPathToFirstSubvol_LVL%d_", i))
-		if err := os.Mkdir(name, 0777); err != nil {
-			t.Fatalf("Failed to create dir %s/%s: %v", subvoldir, name, err)
-		}
-		if err := os.Chdir(name); err != nil {
-			t.Fatal(err)
-		}
+		assert.NilError(t, os.Mkdir(name, 0777), "Failed to create dir %s/%s", subvoldir, name)
+		assert.NilError(t, os.Chdir(name))
 		l += len(name)
 		subvoldir = path.Join(subvoldir, name)
 	}
 
-	if err := subvolSnapshot("", subvoldir, "subvolLVL1"); err != nil {
-		t.Fatal(err)
-	}
-	if err := subvolSnapshot("", subvoldir+"/..", "subvolLVL1_0"); err != nil {
-		t.Fatal(err)
-	}
-	if err := os.Chdir("subvolLVL1"); err != nil {
-		t.Fatal(err)
-	}
+	assert.NilError(t, subvolSnapshot("", subvoldir, "subvolLVL1"))
+	assert.NilError(t, subvolSnapshot("", subvoldir+"/..", "subvolLVL1_0"))
+	assert.NilError(t, os.Chdir("subvolLVL1"))
 	subvoldir = path.Join(subvoldir, "subvolLVL1")
-	if err := subvolSetPropRO(subvoldir, false); err != nil {
-		t.Fatal(err)
-	}
+	assert.NilError(t, subvolSetPropRO(subvoldir, false))
 
 	i := 1
 	for l := 0; l < syscall.PathMax; i++ {
 		name := getMaxFilenameFormPattern(fmt.Sprintf("LongPathToNestedSub_1_LVL%d_", i))
-		if err := os.Mkdir(name, 0777); err != nil {
-			t.Fatalf("Failed to create dir %s/%s: %v", subvoldir, name, err)
-		}
-		if err := os.Chdir(name); err != nil {
-			t.Fatal(err)
-		}
+		assert.NilError(t, os.Mkdir(name, 0777), "Failed to create dir %s/%s", subvoldir, name)
+		assert.NilError(t, os.Chdir(name))
 		l += len(name)
 		subvoldir = path.Join(subvoldir, name)
 	}
 
-	if err := subvolSnapshot("", subvoldir, "subvolLVL2_1"); err != nil {
-		t.Fatal(err)
-	}
+	assert.NilError(t, subvolSnapshot("", subvoldir, "subvolLVL2_1"))
 
 	for ; i > 1; i-- {
-		if err := os.Chdir(".."); err != nil {
-			t.Fatal(err)
-		}
+		assert.NilError(t, os.Chdir(".."))
 		subvoldir = path.Dir(subvoldir)
 	}
 
 	for i, l := 1, 0; l < syscall.PathMax*2; i++ {
 		name := getMaxFilenameFormPattern(fmt.Sprintf("LongPathToNestedSub_2_LVL%d_", i))
-		if err := os.Mkdir(name, 0777); err != nil {
-			t.Fatalf("Failed to create dir %s/%s: %v", subvoldir, name, err)
-		}
-		if err := os.Chdir(name); err != nil {
-			t.Fatal(err)
-		}
+		assert.NilError(t, os.Mkdir(name, 0777), "Failed to create dir %s/%s", subvoldir, name)
```

Comment:
in Go, error message already contains a file name, so you can omit a custom error message entirely, ie
```diff
-		assert.NilError(t, os.Mkdir(name, 0777), "Failed to create dir %s/%s", subvoldir, name)
+		assert.NilError(t, os.Mkdir(name, 0777)

and the error will be  something like

`a_test.go:12: assertion failed: error is not nil: mkdir /some/file: no such file or directory`

which is good enough